napari/components/viewer_model.py

```diff
@@ -1310,8 +1310,20 @@ def open_sample(
         with layer_source(sample=(plugin, sample)):
             if callable(data):
                 added = []
-                for datum in data(**kwargs):
-                    added.extend(self._add_layer_from_data(*datum))
+                needs_error = True
+                for datum in ensure_list_of_layer_data_tuple(data(**kwargs)):
```

Comment:
> to check for an iterable (and update the input typing), and keep everything else the same?

String also is iterable. 

My first thought was to do this:

```suggestion
                for datum in ensure_list_of_layer_data_tuple(list(data(**kwargs))):
```

But it adds additional coping of list.

Reply:
I don't think it's a big problem, these won't ever be such huge lists that it will matter (and if they are, napari will have performance issues elsewhere before this ever becomes a problem :sweat:)

Reply:
Yeah agreed, this will be the simplest fix although I agree that overall the `ensure_list_of_layer_data_tuple` function is maybe too stringent